src/AstProgram.h

```diff
@@ -18,6 +18,7 @@
 #pragma once
 
 #include "AstComponent.h"
+#include "AstFunctorDeclaration.h"
```

Comment:
This could be a forward declaration, as for AstClause/etc. Then #include in the cpp where it's used